provisioning/transport/http/devdoc/http_requirements.md

```diff
@@ -0,0 +1,89 @@
+# azure-iot-provisioning-http Requirements
+
+## Overview
+This module provides HTTP protocol support to communicate with the Azure device provisioning service
+
+```js
+'use strict';
+
+export class Http extends EventEmitter implements DeviceProvisioningTransport {
+  api-version: string;
+  operationPollingIntervalInSeconds: Number
+  constructor(idScope: string, registrationId: string) : void;
+  register(authorization?: string, forceRegistration?: boolean, done(err: Error, result?: Any) => void) : void;
+}
+```
+
+## Public Interface
+
+### constructor(idScope: string, registrationId: string) : void
+The `constructor` creates an Http transport object used to communicate with the Azure device provisioning service
+
+**SRS_NODE_PROVISIONING_HTTP_18_001: [** The `Http` constructor shall accept the following properties:
+- `idScope` - a string specifiying the scope of the provisioning operations,
+- `registrationId` - the registration id for the specific device 
+**]**
+
+
+### register(authorization?: string, forceRegistration?: boolean, callback (err: Error, result?: Any) => void) : void;
+`register` calls into the service to register the given device with the provisioning service.
+
+**SRS_NODE_PROVISIONING_HTTP_18_002: [** `register` shall throw a `ReferenceError` if `callback` is falsy. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_004: [** The registration request shall use Http 1.1. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_005: [** The registration request shall include the current `api-version` as a URL query string value. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_006: [** The registration request shall specify the following in the Http header:
+  Accept: application/json
+  Content-Type: application/json; charset=utf-8 **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_007: [** If an `authorization` string is specifed, the registration request shall include this in the Http header. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_008: [** If `forceRegistration` is specified, the registration request shall include this as a query string value. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_009: [** `register` shall shall POST the registration request to 'https://global.azure-devices-provisioning.net/{idScope}/registrations/{registrationId}' **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_010: [** `register` shall wait for a registration response. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_011: [** If the registration request times out, `register` shall call the `callback` with `TimeoutError` error **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_012: [** If the registration response contains a body, `register` shall deserialize this into a JSON object. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_013: [** If registration response body fails to deserialize, `register` will call the `callback` with a `SyntaxError` error. **]**
```

Comment:
>SyntaxError [](start = 137, length = 11)

- does this error already exist in the SDK? 
- shouldn't we throw here? means it's a programmer error somewhere, likely in the SDK? I don't want to give the customer the impression that this is something they are responsible for by giving them the error in a callback when in fact we are.

Reply:
not convinced by InternalServerError either. might be us screwing up somewhere. I'm not against creating a specific error for that but blaming it on the server doesn't seem right. throwing seems right though

---
In reply to: [140061844](https://github.com/Azure/azure-iot-sdk-node/pull/86#discussion_r140061844) [](ancestors = 140061844)